pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -117,16 +118,25 @@ func (r *ReferenceAdapter) EnsureProjectClaimUpdated() (gcpv1alpha1.ClaimStatus,
 
 	if r.projectClaim.Spec.GCPProjectID == "" {
 		r.projectClaim.Spec.GCPProjectID = r.projectReference.Spec.GCPProjectID
-		err := r.kubeClient.Update(context.TODO(), r.projectClaim)
+	}
+
+	if len(r.projectClaim.Spec.AvailibilityZones) <= 0 {
+		err := r.checkAvailibilityZones()
 		if err != nil {
-			r.logger.Error(err, "Error updating ProjectClaim GCPProjectID")
+			r.logger.Error(err, "Error updating chceking availibility zones")
 			return r.projectClaim.Status.State, err
 		}
 	}
 
+	err := r.kubeClient.Update(context.TODO(), r.projectClaim)
```

Comment:
I think we should only call update if the `ProjectClaim` actually changed. Therefore, the functions called should inform the caller about that.